gematria/datasets/find_accessed_addrs_exegesis.cc

```diff
@@ -134,21 +136,49 @@ Expected<ExecutionAnnotations> ExegesisAnnotator::findAccessedAddrs(
   BenchCode.Key.MemoryValues["memdef1"] = MemVal;
 
   const llvm::MCRegisterInfo &MRI = State.getRegInfo();
+  const std::vector<unsigned> UsedRegisters = gematria::getUsedRegisters(
+      *DisInstructions, State.getRegInfo(), State.getInstrInfo());
+  MemAnnotations.mutable_accessed_blocks()->Reserve(
+      BenchCode.Key.MemoryMappings.size());
 
-  for (unsigned i = 0;
-       i < MRI.getRegClass(X86::GR64_NOREX2RegClassID).getNumRegs(); ++i) {
-    RegisterValue RegVal;
-    RegVal.Register =
-        MRI.getRegClass(X86::GR64_NOREX2RegClassID).getRegister(i);
-    RegVal.Value = APInt(64, kInitialRegVal);
-    BenchCode.Key.RegisterInitialValues.push_back(RegVal);
-  }
+  for (unsigned RegisterIndex : UsedRegisters) {
+    // Skip all the segment registers other than FS and GS as we cannot set
+    // them from userspace and the machine code verifier does not complain if
+    // they are not defined.
```

Comment:
Is there a register class covering the segment registers? There might even be one covering the user-assignable registers `FS` and `GS` or the hard-coded ones. This is an architecture-defined thing (not LLVM-defined or ABI-defined) so there might very likely be one.

Reply:
There's one for the segment registers, but not one that explicitly excludes FS/GS.